### ISSUE TITLE
Parameter explicit_resource_group_name not honored

### ISSUE THREAD

**Issue by reijop** [test-kitchen/kitchen-azurerm#80]
Set the parameter in my .kitchen.yml, and lo, I still see the generated resource-group names.

``` .kitchen.yml
driver:
  name: azurerm

driver_config:
  subscription_id: <%= ENV['AZURE_SUBSCRIPTION_ID'] %>
  location: 'West US'
  machine_size: 'Standard_D1'
  explicit_resource_group_name: test-kitchen
```

And I get this:
```
-----> Creating <dotnet-windows2k12R2-c12>...
       Creating Resource Group: kitchen-dotnet-windows2k12R2-c12-20180508T071306
```

**Comment by stuartpreston** [comment]
This did not repro for me on the latest version 0.14.0.  Can you make sure your `.kitchen` folder is empty to ensure no state is persisting from a previous session?  

My configuration for reference:

*kitchen.yml*
```yml
---
driver:
  name: azurerm

driver_config:
  subscription_id: 43706c4e-abcd-abcd-abcd-e64e6bf63ac4
  location: 'West US'
  machine_size: 'Standard_D1'
  explicit_resource_group_name: test-kitchen

verifier:
  name: inspec

transport:
  ssh_key: ~/.ssh/id_kitchen-azurerm

provisioner:
  name: chef_zero

platforms:
  - name: ubuntu-14.04
    driver:
      image_urn: Canonical:UbuntuServer:14.04.4-LTS:latest
      vm_name: trusty-vm
      vm_tags:
        ostype: linux
        distro: ubuntu

suites:
  - name: default
    run_list:
      - recipe[kitchen-azurerm-repro-80::default]
    verifier:
      inspec_tests:
        - test/integration/default
    attributes:
```

*output*
```
C:\projects\test-kitchen\kitchen-azurerm-repro-80 [master +5 ~0 -0 !]> kitchen list
Instance             Driver   Provisioner  Verifier  Transport  Last Action    Last Error
default-ubuntu-1404  Azurerm  ChefZero     Inspec    Ssh        <Not Created>  <None>
C:\projects\test-kitchen\kitchen-azurerm-repro-80 [master +5 ~0 -0 !]> kitchen create
-----> Starting Kitchen (v1.21.2)
-----> Creating <default-ubuntu-1404>...
       Creating Resource Group: test-kitchen
       Creating deployment: deploy-9efb852037095417
       Adding public key from C:/Users/spreston/.ssh/id_kitchen-azurerm.pub to the deployment.
       Resource Microsoft.Resources/deployments 'pid-18d63047-6cdf-4f34-beed-62f01fc73fc2' provisioning status is Running       Resource Microsoft.Network/virtualNetworks 'vnet' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'trusty-vm' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'trusty-vm' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'trusty-vm' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'trusty-vm' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'trusty-vm' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'trusty-vm' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'trusty-vm' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'trusty-vm' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'trusty-vm' provisioning status is Running
       Resource Template deployment reached end state of 'Succeeded'.
       IP Address is: 104.210.49.65 [kitchen-9efb852037095417.westus.cloudapp.azure.com]
       Finished creating <default-ubuntu-1404> (2m46.10s).
-----> Kitchen is finished. (2m57.57s)

```

**Comment by reijop** [comment]
Yep.  That unblocked me and solved this issue.  Closing as fixed.

(but I'm about to file a new one for another issue ;)
